### PR TITLE
DAOS-1508 test: Validate agent/dmg client configuration features

### PR DESCRIPTION
**Pull request by amjustin13**
As part of this patch two new test files have been added:
daos_agent_config.py
daos_control_config.py

Test cases to validate parsing of configuration file for the agent and dmg . This test is mostly to do negative testing by providing unexpected inputs in combination to the config file.

### CODE REVIEW

File: src/tests/ftest/control/daos_agent_config.yaml [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+  cert_num:
+    config_val:
+       - "cert"
+       - 12345678
+       - "PASS"
+  cert_file_does_not_exist:
+    config_val:
+       - "cert"
+       - /tmp/does_not_exist.ext
+       - "PASS"
```
Comment (shimizukko) [inline src/tests/ftest/control/daos_agent_config.yaml:113]: I'm curious. Isn't this supposed to fail?

File: src/tests/ftest/control/daos_agent_config.yaml [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  runtime_dir_list:
+    config_val:
+      - "runtime_dir"
+      - [12345, ABDCD]
+      - "FAIL"
+  log_file_does_not_exist:
+    config_val:
+      - "log_file"
+      - /does/not/exist.log
+      - "PASS"
```
Comment (shimizukko) [inline src/tests/ftest/control/daos_agent_config.yaml:153]: Isn't this supposed to fail?

File: src/tests/ftest/control/daos_control_config.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+from apricot import TestWithServers
+from command_utils_base import CommandFailure
+
+
+class DaosControlConfigTest(TestWithServers):
```
Comment (shimizukko) [inline src/tests/ftest/control/daos_control_config.py:30]: Why is this test called DaosControlConfigTest? You're checking if dmg storage scan works with given config file, so isn't it more like DmgConfigTest?

File: src/tests/ftest/control/daos_agent_config.yaml (hunk 1 above)
Comment (amjustin13) [reply src/tests/ftest/control/daos_agent_config.yaml:113]: No, it creates a file if it's not there.

File: src/tests/ftest/control/daos_agent_config.yaml (hunk 2 above)
Comment (amjustin13) [reply src/tests/ftest/control/daos_agent_config.yaml:153]: Same as above.

File: src/tests/ftest/control/daos_agent_config.yaml (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/control/daos_agent_config.yaml:113]: I see.

File: src/tests/ftest/control/daos_control_config.py (hunk 3 above)
Comment (amjustin13) [reply src/tests/ftest/control/daos_control_config.py:30]: That's a good point. I called it control config test to keep consistency with the dmg config name which is daos_control.yaml

Comment (shimizukko) [reply src/tests/ftest/control/daos_control_config.py:30]: I see. I didn't know that was the name of the config file.

Comment (mjmac) [reply src/tests/ftest/control/daos_control_config.py:30]: Yeah, FYI... The file was renamed from `daos.yml` to `daos_control.yml` in order to better reflect its relationship to what it's configuring. The dmg tool uses the Control API, but the expectation is that the new API may be used to build other things.

File: src/tests/ftest/control/daos_control_config.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    """Test Class Description:
+    Simple test to verify that the daos_control starts/stops properly given
```
Comment (mjmac) [inline src/tests/ftest/control/daos_control_config.py:32]: Nit: copy-paste comment doesn't quite make sense.

File: src/tests/ftest/control/daos_control_config.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    positive and negative values to its configuration file.
+    :avocado: recursive
+    """
+
+    def test_daos_control_config_basic(self):
+        """
+        JIRA ID: DAOS-1508
+
+        Test Description: Test daos_control start/stops properly.
```
Comment (mjmac) [inline src/tests/ftest/control/daos_control_config.py:41]: also here

File: src/tests/ftest/control/daos_control_config.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        c_val = self.params.get("config_val", "/run/control_config_val/*/")
+
+        # Identify the attribute and modify its value to test value
+        self.assertTrue(
+            self.server_managers[-1].dmg.set_config_value(c_val[0], c_val[1]),
+            "Error setting the '{}' config file parameter to '{}'".format(
+                c_val[0], c_val[1]))
+
+        # Setup the access points with the server hosts
+        self.log.info("Starting dmg config with %s = %s", c_val[0], c_val[1])
```
Comment (mjmac) [inline src/tests/ftest/control/daos_control_config.py:56]: Nit: it's not really "starting" like a daemon... Suggest "Executing dmg with config" or something similar.

**Comment by amjustin13** [comment]
@mjmac , can you take a look at landing? The failures on this build are of known tickets:
degraded-mode test and DAOSCoreTest failure: https://jira.hpdd.intel.com/browse/DAOS-4537
daos_core_test_rebuild test: https://jira.hpdd.intel.com/browse/DAOS-4700

Passed runs of the tests I added are here:
- https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-2667/5/testReport/(root)/DaosAgentConfigTest/
- https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-2667/5/testReport/(root)/DaosControlConfigTest/

File: src/tests/ftest/control/daos_control_config.py (hunk 4 above)
Comment (amjustin13) [reply src/tests/ftest/control/daos_control_config.py:32]: Will do.

File: src/tests/ftest/control/daos_control_config.py (hunk 5 above)
Comment (amjustin13) [reply src/tests/ftest/control/daos_control_config.py:41]: Will do.

File: src/tests/ftest/control/daos_control_config.py (hunk 6 above)
Comment (amjustin13) [reply src/tests/ftest/control/daos_control_config.py:56]: Will do.

**Comment by amjustin13** [comment]
Updated patch fixing the few feedback items.

File: src/tests/ftest/control/daos_control_config.py (hunk 3 above)
Comment (shimizukko) [reply src/tests/ftest/control/daos_control_config.py:30]: I see. Thanks.

**Comment by amjustin13** [comment]
@mjmac, can we land? 
Test failures known flaky tickets:
- 2 failures on pool_svc and 1 failure on cart_self_test tests: https://jira.hpdd.intel.com/browse/DAOS-3945
- 1 ior_small failure, 1 mdtest_small failure and 4 fio_small failures: https://jira.hpdd.intel.com/browse/DAOS-4850

**Pull request merged**